### DISCUSSION TITLE
application related bill of materials

### DISCUSSION THREAD

**Discussion by balasu** [rad-security/kbom#63] [Ideas]
Hi
thanks for an awesome tool, any plans for including an option to extract k8s resources bom like

Namespace,servicename, type(pod,deployment,statefulset,daemonset),containername, image, capacity

**Comment by mateuszdyminski** [comment]
Hi @balasu,

I think our tool can generate all of these things. 

In the attachment you can find example `kbom` file created for my local `kind` cluster. 
[kbom-7e644550-2023-07-05-10-21-20.json.txt](https://github.com/ksoclabs/kbom/files/11955634/kbom-7e644550-2023-07-05-10-21-20.json.txt)

As you can see - there are all k8s resources - including even custom CRDs. 

Do you think something is missing there? We are more than happy to include more info if it's only possible.

**Comment by balasu** [comment]
Hi
Thanks I am just using kbom generate --output=file. could u share me the command tht is used to generate ur txt file.
I think thts enough info on the txt file u have shared

**Reply by mateuszdyminski** [reply]
My file was generated in the following way:

```
kbom generate -o file
```

With kbom in version: `v0.2.0`

I added `.txt` suffix at the end of the filename because of github - it doesn't allow to add `json` file in the attachment of the comment :)


If there anything else we can do for you - just let us know. Cheers!

**Reply by balasu** [reply]
Ok thanks  Mateus, Kbom generates container images, pod names, namespace names,config map,secrets,endpoints,service however it doesn't have a  corelation in the json output. atleast for non kube namespaces probably a different option in kbom command to generate output. like container images, pod names, pod  capacity, namespace names,config map,secrets,endpoints,service . right now there is no straight way to generate this.pl  let me know ur thought

**Reply by mateuszdyminski** [reply]
Probably you are asking about the hierarchy and dependencies - Am I right?

This might be really nice to see the hierarchy and know that particular Image is run in the container, then Pod, ReplicaSet, Deployment and Node. 
Also dependencies and relationships like Service to Pod etc.

Let's discuss that internally as it might be tricky and requires more work. Thanks!

**Reply by balasu** [reply]
yes thanks Mateus
